### PR TITLE
[Merged by Bors] - feat(FiberedCategory/Fibered): define prefibered categories

### PR DESCRIPTION
**Pull request by callesonne**
We define the notion of prefibered categories, as in SGA 1 VI 6.1, and add some basic API.

Co-authored-by: Paul Lezeau <paul.lezeau@gmail.com>

---

- [x] depends on: #13410
- [x] depends on: #13393
- [x] depends on: #12978

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/CategoryTheory/FiberedCategory/Fibered.lean [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+namespace CategoryTheory
+
+variable {𝒮 : Type u₁} {𝒳 : Type u₂} [Category.{v₁} 𝒮] [Category.{v₂} 𝒳]
+
+/-- Definition of a prefibered category.
+
+See SGA 1 VI.6.1. -/
+class Functor.IsPreFibered (p : 𝒳 ⥤ 𝒮) : Prop where
+  has_pullbacks' {a : 𝒳} {R : 𝒮} (f : R ⟶ p.obj a) : ∃ (b : 𝒳) (φ : b ⟶ a), IsCartesian p f φ
```
Comment (semorrison) [inline Mathlib/CategoryTheory/FiberedCategory/Fibered.lean:43]: Perhaps should be called `exists_isCartesian'` (and similarly for the lemma below)?

Comment (callesonne) [reply Mathlib/CategoryTheory/FiberedCategory/Fibered.lean]: Good suggestion! Fixed :)

**Review by sinhp — APPROVED** [review]
Looks good to me.

**Comment by dagurtomas** [comment]
maintainer merge

**Comment by riccardobrasca** [comment]
Thanks!

bors merge

**Pull request closed without merging**